### PR TITLE
txn: Split up the command module so there is one submodule per command

### PR DESCRIPTION
**Pull request by longfangsong**
Signed-off-by: longfangsong <longfangsong@icloud.com>

### What problem does this PR solve?

As @nrc said in sig-transaction#28, we split up the command module so there is one submodule per command.

### What is changed and how it works?

What's Changed:

Promote the command.rs into a dictionary module and move each command to its own submodule.

### Check List 

Tests 

- Unit test

### CODE REVIEW

**Comment by longfangsong** [comment]
@nrc PTAL

File: src/storage/txn/commands/acquire_pessimistic_lock.rs [hunk 1]
```diff
@@ -0,0 +1,52 @@
+use crate::storage::lock_manager::WaitTimeout;
```
Comment (nrc) [inline src/storage/txn/commands/acquire_pessimistic_lock.rs:1]: Every new file needs the copyright comment.

File: src/storage/txn/commands/cleanup.rs [hunk 2]
```diff
@@ -0,0 +1,33 @@
+use crate::storage::metrics::{self, KV_COMMAND_COUNTER_VEC_STATIC};
+use crate::storage::txn::commands::{Command, CommandExt, TypedCommand};
+use crate::storage::txn::latch::{self, Latches};
+use crate::storage::{Context, TimeStamp};
+use crate::{command, ctx, gen_lock, tag, ts, write_bytes};
+use std::fmt::{self, Debug, Display, Formatter};
+use std::iter;
+use txn_types::Key;
```
Comment (nrc) [inline src/storage/txn/commands/cleanup.rs:8]: You can use fully qualified types in the macros and then you shouldn't need all these imports in each file.

File: src/storage/txn/commands/cleanup.rs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+command! {
+    /// Rollback mutations on a single key.
```
Comment (nrc) [inline src/storage/txn/commands/cleanup.rs:11]: Can you move these comments to inner module comments for each command please?

Comment (longfangsong) [reply src/storage/txn/commands/cleanup.rs:11]: If we move these comments to inner module, these comments won't showup on the document web page built by `cargo doc`.
Does that matter?

Comment (nrc) [reply src/storage/txn/commands/cleanup.rs:11]: They should show up as docs for the module. Do they not?

Comment (longfangsong) [reply src/storage/txn/commands/cleanup.rs:11]: > They should show up as docs for the module. Do they not?

No because thouse modules are not `pub`.

Comment (nrc) [reply src/storage/txn/commands/cleanup.rs:11]: Ah, ok, just leave them as is then.

File: src/storage/txn/commands/acquire_pessimistic_lock.rs [hunk 4]
```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
```
Comment (nrc) [inline src/storage/txn/commands/acquire_pessimistic_lock.rs:1]: Should be 2020

**Pull request merged**